lib/statsd_middleware.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+class StatsdMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    status, headers, response = @app.call(env)
+    params = env['action_dispatch.request.path_parameters']
+    key = "api.external.request#status=#{status},controller=#{params[:controller]},action=#{params[:action]}"
```

Comment:
FYI, these are dogstatsd-style tags, which are supported in prometheus, or at least that's my limited understanding from James.